src/struct_frame/boilerplate/csharp/sdk/SerialAndWebSocketTransports.cs

```diff
@@ -0,0 +1,266 @@
+// WebSocket and Serial transports for C#
+// WebSocket uses NetCoreServer, Serial uses System.IO.Ports
+
+using System;
+using System.Threading.Tasks;
+
+namespace StructFrame.Sdk
+{
+    /// <summary>
+    /// WebSocket transport configuration
+    /// </summary>
+    public class WebSocketTransportConfig : TransportConfig
+    {
+        public string Url { get; set; }
+        public int TimeoutMs { get; set; } = 5000;
+    }
+
+    /// <summary>
+    /// WebSocket Transport using NetCoreServer
```

Comment:
Is there a c# way implementing the class in an actual file but not including it unless we need to. 
We should not have example code just the actual implementation. Maybe structframesdk.cs should not import everything and instead the documentation should specify how to import and use specific packages. The same is true for the cpp interface, where asio.h may not need to be included.